pubnub-gson/src/main/java/com/pubnub/api/PubNub.kt

```diff
@@ -0,0 +1,583 @@
+import com.pubnub.api.endpoints.channel_groups.DeleteChannelGroup
+import com.pubnub.api.endpoints.channel_groups.ListAllChannelGroup
+import com.pubnub.api.endpoints.channel_groups.RemoveChannelChannelGroup
+import com.pubnub.api.endpoints.files.DeleteFile
+import com.pubnub.api.endpoints.files.DownloadFile
+import com.pubnub.api.endpoints.files.GetFileUrl
+import com.pubnub.api.endpoints.files.ListFiles
+import com.pubnub.api.endpoints.files.PublishFileMessage
+import com.pubnub.api.endpoints.files.SendFile
+import com.pubnub.api.endpoints.message_actions.AddMessageAction
+import com.pubnub.api.endpoints.message_actions.GetMessageActions
+import com.pubnub.api.endpoints.message_actions.RemoveMessageAction
+import com.pubnub.api.endpoints.objects_api.channel.GetAllChannelsMetadata
+import com.pubnub.api.endpoints.objects_api.channel.GetChannelMetadata
+import com.pubnub.api.endpoints.objects_api.channel.RemoveChannelMetadata
+import com.pubnub.api.endpoints.objects_api.channel.SetChannelMetadata
+import com.pubnub.api.endpoints.objects_api.members.GetChannelMembers
+import com.pubnub.api.endpoints.objects_api.members.ManageChannelMembers
+import com.pubnub.api.endpoints.objects_api.members.RemoveChannelMembers
+import com.pubnub.api.endpoints.objects_api.members.SetChannelMembers
+import com.pubnub.api.endpoints.objects_api.memberships.GetMemberships
+import com.pubnub.api.endpoints.objects_api.memberships.ManageMemberships
+import com.pubnub.api.endpoints.objects_api.memberships.RemoveMemberships
+import com.pubnub.api.endpoints.objects_api.memberships.SetMemberships
+import com.pubnub.api.endpoints.objects_api.uuid.GetAllUUIDMetadata
+import com.pubnub.api.endpoints.objects_api.uuid.GetUUIDMetadata
+import com.pubnub.api.endpoints.objects_api.uuid.RemoveUUIDMetadata
+import com.pubnub.api.endpoints.objects_api.uuid.SetUUIDMetadata
+import com.pubnub.api.endpoints.presence.GetState
+import com.pubnub.api.endpoints.presence.HereNow
+import com.pubnub.api.endpoints.presence.SetState
+import com.pubnub.api.endpoints.presence.WhereNow
+import com.pubnub.api.endpoints.pubsub.Publish
+import com.pubnub.api.endpoints.pubsub.Signal
+import com.pubnub.api.endpoints.push.AddChannelsToPush
+import com.pubnub.api.endpoints.push.ListPushProvisions
+import com.pubnub.api.endpoints.push.RemoveAllPushChannelsForDevice
+import com.pubnub.api.endpoints.push.RemoveChannelsFromPush
+import com.pubnub.api.v2.callbacks.EventListener
+import com.pubnub.api.v2.callbacks.StatusListener
+import com.pubnub.api.v2.entities.Channel
+import com.pubnub.api.v2.entities.ChannelGroup
+import com.pubnub.api.v2.entities.ChannelMetadata
+import com.pubnub.api.v2.entities.UserMetadata
+import com.pubnub.api.v2.subscriptions.Subscription
+import com.pubnub.api.v2.subscriptions.SubscriptionSet
+import com.pubnub.internal.BasePubNubImpl
+import com.pubnub.internal.PubNubImpl
+import java.io.InputStream
+
+interface PubNub :
```

Comment:
Is this ok that kotlin file is mixed with java files?

Reply:
as long as we don't use symbols generated by Lombok then yes, in the long term it would be great to slowly migrate everything to pure Kotlin